### DISCUSSION TITLE
I would like to use chrome as my webview in M1 Mac is this possible?

### DISCUSSION THREAD

**Discussion by jsoneaday** [tauri-apps/tauri#4591] [Q&A] [answered]
I need a browser that supports the Solana Phantom extension, but this extension is not supported on Safari. Can I switch the browser to Chrome on a mac?

**Comment by FabianLars** [comment, marked as answer]
As said on Discord just now, there are 2 big "No"s here: 1) No, you can't change the webview to chromium, 2) No, you can't use extensions in the webview :/
And we can't do anything about that really.

**Reply by biels** [reply]
I have read the project goals and understand that this project aims to not bundle the browser runtime. However, as time goes on it is becoming much clearer that Safari is becoming the new Internet Explorer. A lot of people use chrome / firefox on their macOs devices to avoid issues and some web APIs are directly not available, in some cases for apple "political" reasons related to protecting the revenue from the app store or other reasons. The fact that people are asking for this means something.

My proposed solution: Maybe it is time to challenge this project "big No" and add an option to bundle and use chromium on mac builds (while continuing to not bundle it in platforms where the native web view is already fine). 

I would say this is keeping many from using tauri instead of electron.

What do you say?

**Reply by FabianLars** [reply]
Let's just say that we're aware of the issues people have with webkit, especially on Linux (i know this discussion is about macOS but Linux has bigger problems). I don't really want to go into more detail on that stuff just yet but i think the hint was obvious :D (btw, no promises about anything!)

Also, Microsoft is working on a macOS port for WebView2 which i'd imagine should be a possible alternative too with hopefully wayyy less work for us (i know i know, Microsoft software on Apple devices 😷 but pick your poison i guess)

> I would say this is keeping many from using tauri instead of electron.

Ignoring what i just said above, personally i'm totally fine with that! imho Tauri doesn't have to be for everyone, "pick the right tool for the job" and Tauri can't ever be the right tool for every single job :)

Oh and not to forget, we're _really_ short on maintainers especially for macOS so we have to be careful about changes this large.

**Reply by infinite2009** [reply]
I cannot agree any more with you bro @biels. Be forced to use safari on Mac as the only option make me struggling. If it's possible to choose anthor webview on Mac will be perfect. There is no doubt that Safari cannot support the best application experience and it will be a large obstacle to get electron developers moving to tauri.

**Comment by Kadrian** [comment]
@jsoneaday have you found another solution here? I'm new to Tauri and I can't find how to switch from safari dev tools to chrome dev tools. Hmm.

**Comment by MikeMitterer** [comment]
Is the answer from above still the state of affairs? 
If so - I think you are ignoring real world problems and needs. Tauri, like Electron, is a write once, run anywhere solution - or at least it positions itself like such. But WebKit on Mac is a showstopper for most Devs out there. The combination Rust + Web is a great approach, but without Chromium on Mac and Linux the whole thing is a nice proof of concept but nothing more. 
Is there hope for Chromium on Mac with Tauri???

**Reply by FabianLars** [reply]
Yes, pretty much. We played with the idea of creating [cef bindings](https://github.com/wusyong/cef-rs) but that's a ton of work, more than we can handle right now. Microsoft is actively working on a port of WebView2, which we use on Windows already and we're hoping that it will be easier to integrate into tauri/wry than cef but we'll have to wait and see.

**Reply by MikeMitterer** [reply]
Very cool and thanks! (CEF - https://github.com/chromiumembedded)

**Reply by SteveMcFarlin** [reply]
@wusyong @FabianLars Apologies that this is out of context. I did not want to open an issue on `cef-rs` to ask a question. Given the amount of work to integrate CEF, and the port MS is doing... do you plan to continue working on `cef-rs`? I work with cef via the [gstcef](https://github.com/centricular/gstcefsrc) project. The reason I am here is WebKitGTK will not work for integrating our application into Tauri. Hence, I am looking as the others have to integrate a chrome backed renderer (WebRTC is also a must). My only target would be Linux.

**Comment by Catchmaker** [comment]
Is this relevant? https://github.com/MicrosoftEdge/WebView2Feedback/issues/1314#issuecomment-1751174077

"New Microsoft Teams claims to use WebView2 now. Does this mean WebView2 is now running on macOS?"

---
Also: I can't help with code but might contribute towards a bounty (if such things are done for Tauri).

**Reply by kurdin** [reply]
It would be very nice if we can use WebView2 on mac at least for development. I think Safari WebKit is ok for prod and final app but Safari dev tools just does not work well in any serious development.  Hope you @FabianLars can find the way to bring MS WebView2 to mac sooner than later. 🙏

**Reply by FabianLars** [reply]
Well, we still have to wait for Microsoft to make it public. Who knows how long that'll take...

**Comment by babakfp** [comment]
I deleted Edge browser with the rest of the related crap. You're telling me I need to install it to use Tauri? 😂 Sounds like a bad joke!

**Reply by FabianLars** [reply]
webview2 is a system component so it should work without edge (on win11 iirc even the start menu uses it, so you won't be able to really remove it). EIther way, if it's missing, and tauri will install just webview2, without edge itself.

**Reply by babakfp** [reply]
After my comment, I deleted everything, including Rust.
There is a folder called `EdgeWebView`, and it has the word `Edge` in it. I don't like that.
Start Menu probably uses that for showing web search results, which is disabled.
If the version that Tauri will install, isn't going to have the word "Edge" in it, then I'm OK with it.

**Reply by simonhyll** [reply]
@babakfp Tauri relies on using the webview provided by the system. On Linux that's webkitgtk, on Mac and iOS that's Safari, on Android it's Chrome, and yes on Windows it's WebView2, which is an Edge based webview.

If that's a deal breaker for you then I guess we're sad to see you go. Check back in on Tauri v3 (possibly v4) when it comes around where our top priority is adding e.g. CEF (Chromium) as an alternative webview

**Comment by repugraf** [comment]
Guys, stop demanding features without any contribution. It’s a free and open source project. If you want some feature really bad then ether open PR and implement it or fork the project and change it to your needs.
If you need chromium just use electron. There is no shame in that. Electron is a stable project that gets thing done.

**Reply by vednig** [reply]
The stability and reliability of rust is also important
